Shared/Objects/FilterDrawerButton.swift

```diff
@@ -0,0 +1,101 @@
+//
+// Swiftfin is subject to the terms of the Mozilla Public
+// License, v2.0. If a copy of the MPL was not distributed with this
+// file, you can obtain one at https://mozilla.org/MPL/2.0/.
+//
+// Copyright (c) 2023 Jellyfin & Jellyfin Contributors
+//
+
+import Defaults
+import Foundation
+
+enum FilterDrawerButton: String, CaseIterable, Defaults.Serializable, Displayable, Identifiable {
+
+    case filters
+    case genres
+    case order
+    case sort
+
+    var displayTitle: String {
+        switch self {
+        case .filters:
+            return L10n.filters
+        case .genres:
+            return L10n.genres
+        case .order:
+            return L10n.order
+        case .sort:
+            return L10n.sort
+        }
+    }
+    
+    var id: String {
+        rawValue
+    }
+
+    var settingsFilter: WritableKeyPath<ItemFilters, [ItemFilters.Filter]> {
```

Comment:
I don't think "settings" in the name fits, let's find a different name.

Reply:
I changed this to "itemFilter" since that is what it returns. To be more in line with that naming, i am using isItemFilterActive for the function and itemFilterDefault for the setting where the default value is assigned on the enum. Feel free to let me know if that naming is confusing!